core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineStrategy.java

```diff
@@ -77,4 +79,8 @@ void dropIndexExecute(Connection connection, String schema, String table, String
   default String encloseFullTableName(String schema, String table) {
     return enclose(schema) + "." + enclose(table);
   }
+
+  SelectQuery buildSelectQuery(SelectQuery.Builder builder, int limit);
+
+  UpsertQuery buildUpsertQuery(UpsertQuery.Builder builder);
```

Comment:
Not sure these methods are proper abstractions. But, I have no specific ideas for now, so let's keep them. And we can revisit it in the future if needed. Thanks.